### ISSUE TITLE
GoTestFunc fail with vendored dependency

### ISSUE THREAD

**Issue by qJkee** [fatih/vim-go#1709]
### What did you do? (required. The issue will be **closed** when not provided.)
Run :GoTestFunc

### What did you expect to happen?
SUCESS

### What happened instead?
Can't find vendored dependency

### Configuration (**MUST** fill this out):

* Vim version (first two lines from `:version`):
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Feb 19 2018 13:11:19)
macOS version

* Go version (`go version`):
go version go1.10 darwin/amd64

* Go environment (`go env`):
GOARCH="amd64"
GOBIN=""
GOCACHE="/Users/aaa/Library/Caches/go-build"
GOEXE=""
GOHOSTARCH="amd64"
GOHOSTOS="darwin"
GOOS="darwin"
GOPATH="/Users/a/Documents/Go"
GORACE=""
GOROOT="/usr/local/Cellar/go/1.10/libexec"
GOTMPDIR=""
GOTOOLDIR="/usr/local/Cellar/go/1.10/libexec/pkg/tool/darwin_amd64"
GCCGO="gccgo"
CC="clang"
CXX="clang++"
CGO_ENABLED="1"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fno-caret-diagnostics -Qunused-arguments -fmessage-length=0 -fdebug-prefix-map=/var/folders/96/zcl8qxhn1p79db631pj1ypxc0000gn/T/go-build667610832=/tmp/go-build -gno-record-gcc-switches -fno-common"

* vim-go version:
  latest

* `vimrc` you used to reproduce (use a *minimal* vimrc with other plugins disabled; do not link to a 2,000 line vimrc):

**Comment by Carpetsmoker** [comment]
Some details on your code would be helpful. Just "Can't find vendored dependency" is not a very useful problem description.

See e.g. https://stackoverflow.com/help/mcve

**Comment by qJkee** [comment]
It returns following message.
But if i run `:GoBuild`, or `go build ` in terminal, all is ok
|19 col 2| cannot find package "github.com/markbates/goth/gothic" in any of:
package "github.com/markbates/goth/gothic" in vendor directory

**Comment by Carpetsmoker** [comment]
Okay ... but that still doesn't give me a whole lot of information about how I can reproduce this problem so it can be fixed. `:GoTestFunc` – and all other commands – should work fine with vendored packages. So either you're doing something special which is triggering an obscure bug, or you made a mistake somewhere which is causing the error.

Without any information at all (such as the code you used, more detailed information about your GOPATH/directory layout, the *exact* commands you issued, your vimrc, etc.) it's going to be very hard to give any meaningful response beyond "it works for me".

Put yourself in my position: how would you know what is going wrong on a stranger's system without any information? Would *you* be able to even make a guess about what's going wrong based on *just the information in this issue*? Probably not, and I'm not smarter than you, so neither can I :-)

**Comment by qJkee** [comment]
Example =)

I have file etcdStorage.go, and test file called etcdStorage_test.go

When, inside etcdStorage.go i run `:GoTest`, i get following message
```
kvstorage/etcdStorage.go|10| 2: cannot find package "github.com/coreos/etcd/clientv3" in any of:
~                              |  2 ||      /usr/local/Cellar/go/1.10/libexec/src/github.com/coreos/etcd/clientv3 (from $GOROOT)
~                              |  3 ||      /Users/s/Documents/Go/src/github.com/coreos/etcd/clientv3 (from $GOPATH)
```
but this package is in vendor foldier( building is file)
MAC OS high sierra
vim-go - latest
vim - 8.0

**Comment by bhcleek** [comment]
What's the full path of the directory that contains those two files?

**Comment by qJkee** [comment]
`$GOPATH/src/foo/kvstorage` <- here is 2 files(ectdStorage.go and etcdStorage_test.go)
and vendor directory for this project is
`$GOPATH/src/foo/vendor`

**Comment by genzj** [comment]
@qJkee I may encounter the same problem. does `go list github.com/markbates/goth/gothic` under the project path (which contains the `vendor` folder) return the correct package path in your situation?

**Comment by fatih** [comment]
@genzj can you please open a test repository on github with a basic code example and README.md that describes how to reproduce the issue. Unfortunately they way you describe is not helpful for us. Thanks

**Comment by genzj** [comment]
@fatih thanks for the reply. I opened two repos trying to make the situation clear.

1. say https://github.com/genzj/vim-go-vendor-dep is the dependency I'm going to include in my project, and
2. https://github.com/genzj/vim-go-vendor-proj is my working project.

but for the 2nd repo, I manage the dependencies with vendor folder, or more precisely [glide](https://github.com/Masterminds/glide).

now with vim-go installed, in gvim, trying to include the package `dep` of `github.com/genzj/vim-go-vendor-dep` like `GoImportsAs dep github.com/genzj/vim-go-vendor-dep`, encounters error: `Can't find import: github.com/genzj/vim-go-vendor-dep`.

I tracked the error message and noticed it's caused by error retcode of `go list github.com/genzj/vim-go-vendor-dep`. A surprise to me that `go list` ignores vendor folder. Nevertheless, `goimports` and (of course) `govendor list` works fine.

I made a small patch which solved my problem by calling `govendor list` besides `go list`. I'm going to create a PR for your reference once I figure out why auto complete still not working.
